api/controllers/arrive.js

```diff
@@ -79,321 +79,310 @@ module.exports = {
 
     // Find the user who is currently logged in, and thus arriving in the zone.
     // (This is different between the web app and the mobile app-- hence our helper.)
-    sails.helpers.checkAuth({req:req})
-    .exec({
-      error: function(err) { return exits.error(err); },
-      notAuthenticated: function() { return exits.notAuthenticated(); },
-      success: function(loggedInUserId) {
-
-        User.findOne({ id: loggedInUserId })
-        .exec(function(err, thisUser){
-          if (err) { return exits.error(err); }
-          if (!thisUser) { return exits.error(new Error('The requesting user is logged in as user `'+loggedInUserId+'`, but no such user exists in the database.  This should never happen!')); }
-
-          if (thisUser.currentZone) {
-            // Unsubscribe this user from the old zone, and publish her departure
-            // to update other clients' UIs.
-            try {
-              Zone.unsubscribe(req, [thisUser.currentZone]);
-
-              Zone.publish([thisUser.currentZone], {
-                verb: 'userLeft',
-                username: thisUser.username
-              }, req);
-
-            } catch (e) { return exits.error(e); }
+    // > Note that you could also just as easily put this kind of "yes/no" check in
+    // > a policy.  We're just avoiding that here to keep this example as simple as
+    // > "magic-free" as possible.
+    var loggedInUserId;
```

Comment:
@ juliandavidmr note that this kind of block could be very much simplified now thanks to [`.intercept()`](https://next.sailsjs.com/documentation/reference/waterline-orm/queries/intercept) and the use of ["special exit signals"](https://next.sailsjs.com/documentation/concepts/actions-and-controllers#?exit-signals)

e.g. instead of the `try`/`catch` block, this becomes a one-liner:

```js
var loggedInUserId = await sails.helpers.checkAuth({req}).intercept('notAuthenticated', 'notAuthenticated');
```